src/containers/container.js

```diff
@@ -7,25 +7,52 @@ export default class Container extends React.Component {
   constructor(props) {
     super(props);
     this.handleOnClick = this.handleOnClick.bind(this);
-    this.state = {selectedNoteId: null};
+    this.state = {selectedNoteId: null, notes: [...this.props.data]};
+    this.handleAddNewNote = this.handleAddNewNote.bind(this);
+    this.onUpdateNote = this.onUpdateNote.bind(this);
   }
 
   handleOnClick(id) {
     this.setState({selectedNoteId: id});
   }
 
+  handleAddNewNote(){
+    const currentNotes = this.state.notes;
+    const id = new Date().getTime();
+    this.setState({notes: [
+      {
+        title: "",
+        content: "",
+        id,
+      },
+      ...currentNotes
+    ],
+    selectedNoteId: id,
+    })
+  }
+
+  onUpdateNote(value, id, type) {
+    const notes = this.state.notes;
+    const updatingNoteIndex = notes.findIndex(note => note.id === id);
+    notes[updatingNoteIndex][type] = value;
```

Comment:
Avoid mutate directly to an item of current state, this is anti-pattern. Create a new one copy is the correct way.
More details [here](https://reactjs.org/docs/optimizing-performance.html#examples)
Eg: `const notes = [...this.state.notes]`